woocommerce/woocommerce-ios
# Dashboard: handle error when when user has no permission to view site visit stats

Fixes #823 

## Changes
- In `DotcomError`, added the following two cases. Unfortunately, our dotcom [API error identifier is not unique](https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/) for each error scenario so there is code to check the error message
  - `noStatsPermission`: when the user has no permission to view site stats
  - `statsModuleDisabled`: when Jetpack site stats module disabled
- Added `noPermission` error case to `SiteVisitStatsStoreError`, and mapped the new `DotcomError` cases (in the Networking layer) to the corresponding `SiteVisitStatsStoreError` cases (in the Yosemite layer)
- Handled `noPermission` site visit stats error case the same as when stats module is disabled (https://github.com/woocommerce/woocommerce-ios/pull/1110 https://github.com/woocommerce/woocommerce-ios/pull/1105) - hiding visitor stats UI without alert toast
- Updated unit tests

## Testing
- Prerequisite: have a store with Jetpack installed, with another store user of `Shop Manager` role (could be any other role than `admin`, and the following steps use `Shop Manager` as the example)
- Go to Jetpack modules settings page at your store's `/wp-admin/admin.php?page=jetpack_modules`, and find the "site stats" module. Make sure the site stats module is activated
- Go to your store's `/wp-admin/admin.php?page=jetpack#/traffic`, expand the "site stats" section, and make sure `Shop manager` is turned off under `Allow stats reports to be viewed by` subsection
- Launch the app with the store --> there should not be a generic alert as in the screenshot [here](https://github.com/woocommerce/woocommerce-ios/issues/932#issuecomment-488332137) and the visitor stats should be hidden
- Go back to your store's `/wp-admin/admin.php?page=jetpack#/traffic`, "site stats" section, and turn on `Shop manager` under `Allow stats reports to be viewed by` subsection
- Go back to the app with the store and pull down to refresh on the Dashboard --> visitor stats should be shown again
- You can go to Jetpack modules settings page again at your store's `/wp-admin/admin.php?page=jetpack_modules`, and deactivate "site stats" module. Going back to the app, the visitor stats should be hidden again (for the reason of stats module disabled) without alert toast

Update release notes:
